### DISCUSSION TITLE
'Microsoft.NETCore.App', version '3.1.0' Not found

### DISCUSSION THREAD

**Discussion by cthredgold-quativa** [community/community#40475] [Actions]
### Select Topic Area

Bug

### Body

Hey Team,

We're seeing an issue with our action that Builds and runs our tests.  The error that is coming back is that `'Microsoft.NETCore.App', version '3.1.0'` is not found.  Below is a snippet of the output during the unit test step. 

```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Testhost process for source(s) '/home/runner/work/Fenix_backend/Fenix_backend/tests/QuaTiVa.Integrations.Monday.Tests/bin/Debug/netcoreapp3.1/QuaTiVa.Integrations.Monday.Tests.dll' exited with error: You must install or update .NET to run this application.
App: /home/runner/.nuget/packages/microsoft.testplatform.testhost/17.2.0/lib/netcoreapp2.1/testhost.dll
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '3.1.0' (x64)
.NET location: /usr/share/dotnet/
The following frameworks were found:
  6.0.11 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  7.0.0 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed
To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.0&arch=x64&rid=ubuntu.22.04-x64
. Please check the diagnostic logs for more information.

Test Run Aborted.
     1>Done Building Project "/home/runner/work/Fenix_backend/Fenix_backend/quativa-api.sln" (VSTest target(s)) -- FAILED.
```

Our actions yml looks like this 

```name: .NET

on:
  push:
    branches: [ "master", "staging", "dev" ]
  pull_request:
    branches: [ "master", "staging", "dev" ]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v3
    - name: Setup .NET
      uses: actions/setup-dotnet@v3
      with:
        dotnet-version: 6.0.x
    - name: Restore dependencies
      run: dotnet restore
    - name: Build
      run: dotnet build --no-restore
    - name: Test
      run: dotnet test --no-build --verbosity normal
```

If I change the `dotnet-version` to `3.1.0` It won't build giving this error. 

```
dotnet-install: Attempting to download using primary link https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.0/dotnet-sdk-3.1.0-linux-x64.tar.gz
curl: (22) The requested URL returned error: 404
dotnet-install: The resource at primary link 'https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.0/dotnet-sdk-3.1.0-linux-x64.tar.gz' is not available.
dotnet-install: Attempting to download using primary link https://dotnetbuilds.azureedge.net/public/Sdk/3.1.0/dotnet-sdk-3.1.0-linux-x64.tar.gz
curl: (22) The requested URL returned error: 404
dotnet-install: The resource at primary link 'https://dotnetbuilds.azureedge.net/public/Sdk/3.1.0/dotnet-sdk-3.1.0-linux-x64.tar.gz' is not available.
dotnet_install: Error: Could not find `.NET Core SDK` with version = 3.1.0
```

If I had to hazard a guess, the blob was taken down because support for 3.1.0 is going away next month.  But thought I'd ask in case anyone can see if there's something I'm missing. 

Thanks,

Cameron Thredgold
Quativa

**Comment by ossanna16** [comment]
Hi there @cthredgold-quativa and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by Robert-96** [comment]
Hi, 

If you want to use version .NET SDK `3.1.0` you should set `dotnet-version` to `3.1.100`, or you can use  `3.1.x` or `3.1` to install the latest patch version of .NET SDK on the channel `3.1`.

You can find more about how `dotnet-version` works here: https://github.com/actions/setup-dotnet#supported-version-syntax
